### PR TITLE
SimpleSchema.Type.DateTime can be nullable

### PR DESCRIPTION
**Pull request by kenichirow**
Pass a nullable: true as an option in SimpleSchema.Type.DateTime,  It will raise an error.

```elixir
defmodule MyStruct do
    defstruct [:id, :datetime]

    @behaviour SimpleSchema

    @impl SimpleSchema
    def schema([]) do
      %{
        id: :integer,
        datetime: {SimpleSchema.Type.DateTime, nullable: true}
      }
    end
end
```

```elixir
[
 %{
    id: 1,
    datetime: nill
  }
]
```
```
> SimpleSchema.from_json(MyStruct, input)
** (RuntimeError) failed from_json/2: [{"Type mismatch. Expected String but got Null.", "#/0/start_at"}, {"Type mismatch. Expected String but got Null.", "#/1/start_at"}]
```

It seems to be caused by SimpleSchema.Type.DateTime.schema / 1 discarding `opts`

```elixir
def schema(_opts) do
   {:string, format: :datetime}
end
```

 So I fix it as follows.

```elixir
def schema(opts) do
    merged_opts = Keyword.merge(opts, format: :datetime)
    {:string, merged_opts}
end
```

### CODE REVIEW

File: lib/simple_schema/type/date_time.ex [hunk 1]
```diff
@@ -2,20 +2,34 @@ defmodule SimpleSchema.Type.DateTime do
...
+    do_to_json(schema, value, opts)
   end
 
   @impl SimpleSchema
-  def to_json(_schema, value, _opts) do
+  def from_json(schema, value, opts) do
+    do_from_json(schema, value, opts)
+  end
+
+  defp do_to_json(_schema, nil, _opts), do: {:ok, ""}
```
Comment (melpon) [inline lib/simple_schema/type/date_time.ex]: It should return `{:ok, nil}`

File: test/simple_schema_test.exs [hunk 2]
```diff
@@ -142,6 +142,53 @@ defmodule SimpleSchemaTest do
...
+    }
+
+    normal_output = %MyStruct.Nullable{
+      id: 1,
+      datetime: datetime
+    }
+
+    assert {:ok, null_datetime_output} == SimpleSchema.from_json(MyStruct.Nullable, null_input)
+    assert {:ok, null_datetime_output} == SimpleSchema.from_json(MyStruct.Nullable, omitted_input)
+    assert {:ok, normal_output} == SimpleSchema.from_json(MyStruct.Nullable, normal_input)
```
Comment (melpon) [inline test/simple_schema_test.exs]: Please test to_json with nil value.

File: test/simple_schema_test.exs [hunk 3]
```diff
@@ -34,7 +34,7 @@ defmodule SimpleSchemaTest do
         username: {:string, min_length: 4},
         address: :string,
         internal: MyInternal,
-        datetime: {SimpleSchema.Type.DateTime, optional: true}
+        datetime: {SimpleSchema.Type.DateTime, optional: true, nullable: true}
```
Comment (melpon) [inline test/simple_schema_test.exs]: It seems to do not need.

File: lib/simple_schema/type/date_time.ex (hunk 1 above)
Comment (kenichirow) [reply lib/simple_schema/type/date_time.ex]: Thanks! fixed it

File: test/simple_schema_test.exs (hunk 2 above)
Comment (kenichirow) [reply test/simple_schema_test.exs]: added to_json/2 test.
https://github.com/gumi/simple_schema/pull/2/commits/0e193974e5f260a1749a2e63de69b763ca109e8d

File: test/simple_schema_test.exs (hunk 3 above)
Comment (kenichirow) [reply test/simple_schema_test.exs]: Sorry it was a mistake,  deleted.

**Comment by melpon** [comment]
It is a good feature. Thanks!

**Pull request merged**